test/object-store/sync/flx_sync.cpp

```diff
@@ -3892,16 +3944,17 @@ TEST_CASE("flx: open realm + register subscription callack while bootstrapping",
             };
 
             async_open_task1->start(open_callback_task_1);
+            REQUIRE(subscription_pf.future.get());
```

Comment:
this is now essentially a different test. we now wait until the callback is invoked before starting the second task. we should aim to make it work for the initial test I think.